### PR TITLE
5765 feature specs

### PR DESCRIPTION
**Pull request by X0nic**

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -39,6 +39,9 @@ class Application < Rails::Application
 
     config.action_mailer.default_url_options = { host: ENV['MADELINE_HOSTNAME'] }
   end
+
+  ## beware, for now this seems to be required for proper rendering of all wice_grid views.  (without, view contents is all html escaped)
```
Comment (X0nic) [inline config/application.rb]: Seems all Environments need this line, so I moved it into application.rb

File: spec/features/admin/loan_flow_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+feature 'loan flow' do
+  let(:division) { create(:division) }
+  let(:user) { create(:person, :with_member_access, :with_password, division: division).user }
+  let!(:loan) { create(:loan, division: division) }
+
+  before do
+    login_as(user, scope: :user)
+  end
+
+  # This should work, but for some reason it fails a lot more often
```
Comment (X0nic) [inline spec/features/admin/loan_flow_spec.rb]: This does not make a lot of sense, but specs will fail more often on loan for no reason it seems.

Comment (X0nic) [inline spec/features/admin/loan_flow_spec.rb]: Adding `wait` to the find calls did help, but I think in the end it was a race condition on `.loan-fields` there were actually 2 sections with that class, and the backbone view that was responsible for showing and hiding was attached to that class.

**Comment by X0nic** [comment]
A new error has reared its head @hooverlunch. This one could be related to `DatabaseCleaner` / `FactoryGirl`, or it could be a real bug. Not sure which it is.

It also happens to be sporadic. Probably only every 5 test runs or so. 

```
Failures:

  1) loan flow can index/show/edit and change division
     Got 0 failures and 2 other errors:
     Shared Example Group: :flow called from ./spec/features/admin/loan_flow_spec.rb:13

     1.1) Failure/Error: click_button "Update #{model_to_test.model_name.human}"

          Capybara::Poltergeist::JavascriptError:
            One or more errors were raised in the Javascript code on the page. If you don't care about these errors, you can ignore them by setting js_errors: false in your Poltergeist configuration (see documentation for details).

            ReferenceError: Can't find variable: gvjs_CK
            ReferenceError: Can't find variable: gvjs_CK
                at https://www.gstatic.com/charts/45.1/js/jsapi_compiled_corechart_module.js:1 in global code
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/poltergeist-1.13.0/lib/capybara/poltergeist/browser.rb:376:in `command'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/poltergeist-1.13.0/lib/capybara/poltergeist/browser.rb:192:in `click'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/poltergeist-1.13.0/lib/capybara/poltergeist/node.rb:17:in `command'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/poltergeist-1.13.0/lib/capybara/poltergeist/node.rb:131:in `click'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/capybara-2.6.2/lib/capybara/node/element.rb:134:in `block in click'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/capybara-2.6.2/lib/capybara/node/base.rb:84:in `synchronize'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/capybara-2.6.2/lib/capybara/node/element.rb:134:in `click'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/capybara-2.6.2/lib/capybara/node/actions.rb:39:in `click_button'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/capybara-2.6.2/lib/capybara/session.rb:686:in `block (2 levels) in <class:Session>'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/capybara-2.6.2/lib/capybara/dsl.rb:51:in `block (2 levels) in <module:DSL>'
          # ./spec/support/helpers/feature_spec_helpers.rb:46:in `block (2 levels) in <module:FeatureSpecHelpers>'

     1.2) Failure/Error: division_and_ancestors = division.self_and_ancestors

          ActionView::Template::Error:
            undefined method `self_and_ancestors' for nil:NilClass
          # ./app/policies/application_policy.rb:77:in `check_self_and_ancestors_for_role'
          # ./app/policies/application_policy.rb:65:in `division_member'
          # ./app/policies/application_policy.rb:73:in `division_member_or_admin'
          # ./app/policies/application_policy.rb:16:in `show?'
          # ./app/helpers/admin_helper.rb:23:in `authorized_form_field'
          # ./app/views/admin/loans/_form.html.slim:21:in `block in _app_views_admin_loans__form_html_slim___3072112491025391358_70215442214580'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/simple_form-3.2.1/lib/simple_form/action_view_extensions/form_helper.rb:26:in `block in simple_form_for'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/simple_form-3.2.1/lib/simple_form/action_view_extensions/form_helper.rb:45:in `with_simple_form_field_error_proc'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/simple_form-3.2.1/lib/simple_form/action_view_extensions/form_helper.rb:25:in `simple_form_for'
          # ./app/views/admin/loans/_form.html.slim:1:in `_app_views_admin_loans__form_html_slim___3072112491025391358_70215442214580'
          # ./app/views/admin/loans/_details.html.slim:44:in `_app_views_admin_loans__details_html_slim___3329887229876861961_70215524485600'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/remotipart-1.2.1/lib/remotipart/render_overrides.rb:14:in `render_with_remotipart'
          # ./app/controllers/admin/loans_controller.rb:43:in `show'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/route_translator-4.3.0/lib/route_translator/extensions/action_controller.rb:20:in `set_locale_from_url'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/sql_queries_count-0.0.1/lib/sql_queries_count/controller_runtime.rb:14:in `process_action'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/bullet-5.0.0/lib/bullet/rack.rb:10:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/warden-1.2.6/lib/warden/manager.rb:35:in `block in call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/warden-1.2.6/lib/warden/manager.rb:34:in `catch'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/warden-1.2.6/lib/warden/manager.rb:34:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/etag.rb:24:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/conditionalget.rb:25:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/head.rb:13:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/remotipart-1.2.1/lib/remotipart/middleware.rb:27:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:225:in `context'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:220:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/railties-4.2.6/lib/rails/rack/logger.rb:38:in `call_app'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/railties-4.2.6/lib/rails/rack/logger.rb:20:in `block in call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/railties-4.2.6/lib/rails/rack/logger.rb:20:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/quiet_assets-1.1.0/lib/quiet_assets.rb:27:in `call_with_quiet_assets'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/request_store-1.3.0/lib/request_store/middleware.rb:9:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/methodoverride.rb:22:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/runtime.rb:18:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/lock.rb:17:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/sendfile.rb:113:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/railties-4.2.6/lib/rails/engine.rb:518:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/railties-4.2.6/lib/rails/application.rb:165:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/urlmap.rb:66:in `block in call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/urlmap.rb:50:in `each'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/urlmap.rb:50:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/capybara-2.6.2/lib/capybara/server.rb:19:in `call'
          # /Users/nate/.rvm/gems/ruby-2.2.3@madeline_system/gems/rack-1.6.4/lib/rack/handler/webrick.rb:88:in `service'
          # ------------------
          # --- Caused by: ---
          # NoMethodError:
          #   undefined method `self_and_ancestors' for nil:NilClass
          #   ./app/policies/application_policy.rb:77:in `check_self_and_ancestors_for_role'

Finished in 8.13 seconds (files took 5.96 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/features/admin/loan_flow_spec.rb:3 # loan flow can index/show/edit and change division

Coverage report generated for RSpec to /Users/nate/Documents/dev/sassafras/madeline_system/coverage. 1176 / 1735 LOC (67.78%) covered.
```

File: README.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
 * ruby 2.2.x
 * postgresql
 * mysql (for migrating legacy data)
+* PhantomJS 2.1.x or higher
```
Comment (hooverlunch) [inline README.md]: Nice addition.

File: config/application.rb (hunk 1 above)
Comment (hooverlunch) [inline config/application.rb]: 👍

File: spec/features/admin/division_flow_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+feature 'division flow' do
+
+  let(:division) { create(:division) }
+  let(:user) { create(:person, :with_admin_access, :with_password, division: division).user }
```
Comment (hooverlunch) [inline spec/features/admin/division_flow_spec.rb]: This seems to be repeated a lot. Can we make it its own factory or helper method?

File: spec/features/admin/project_log_flow_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+feature 'loan flow' do
+
+  let(:division) { create(:division) }
+  let(:user) { create(:person, :with_member_access, :with_password, division: division).user }
+  let!(:project_log) do
+    # summary is added in an after create, we need to ensure it is saved first
+    log = create(:project_log, division: division)
+    log.save!
+    log
```
Comment (hooverlunch) [inline spec/features/admin/project_log_flow_spec.rb]: Please use `tap` here. e.g. `create(:project_log, division: division).tap(&:save!)`

File: spec/support/helpers/feature_spec_helpers.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      # Change back to all divisions, and ensure it reloads properly
+      find('[data-expands="division-dropdown"]').click
+      find('.select_division_form').select('All Divisions')
+      expect(find('.madeline')).to have_content 'Madeline'
+      expect(find('[data-expands="division-dropdown"]')).to have_content 'Select Division'
+
+      # Now test index/show/edit
+      expect(page).to have_content(model_to_test.name)
+
+      find("##{model_to_test.model_name.plural}").click_link(model_to_test.id)
```
Comment (hooverlunch) [inline spec/support/helpers/feature_spec_helpers.rb]: `model_to_test.model_name` appears a bunch of times below here. Assign to a shorter var name?

**Pull request merged**